Automattic/jetpack
# [Boost] Fix warning when clearing Critical CSS

When changing theme, if WP_DEBUG is enabled a warning is produced about calling `WP_Object_Cache::delete` incorrectly. It looks like we were passing in `null`, expecting it to clear all cached items in the given group - which is not supported.

This PR fixes this by properly calling `wp_cache_flush_group` on systems where it is supported, otherwise falling back to manually deleting items one-by-one.

## Proposed changes:
* Fix incorrect call to clear wp_cache items.

### Other information:

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* Turn on WP_DEBUG
* Change themes
* Make sure no warnings are produced
* Test both code paths; with and without `wp_cache_flush_group`
